### ISSUE TITLE
Error when trying to disconnect then reconnecting to babashka repl

### ISSUE THREAD

**Issue by tami5** [liquidz/vim-iced#296]
Hey @liquidz 

When running `InstantConnect babashka` then Doing `Disconnect` and   `InstantConnect babashka` I get the following error

```vim
Disconnected.
Connecting...
Error detected while processing function <SNR>138_on_exit[2]..<SNR>139_on_out_close[1]..<lambda>121232[1]..<SNR>136___instant_babashka[7]..iced#util#wait
[10]..<lambda>121234[1]..iced#repl#connect[6]..iced#nrepl#connect[22]..83:
line    6:
connection failed: connection refused

```

Also does reconnect work also for babashka? 

Thanks

**Comment by liquidz** [comment]
@tami5 Thanks for your reporting!

It seems neovim specific problem. (reconnecting works with vim)

**Comment by liquidz** [comment]
@tami5 I tried to fix.
Could you try `dev` branch?

**Comment by tami5** [comment]
On it

**Comment by tami5** [comment]
Reconnect after doing instant connect with babashka and clojure :

```
Error detected while processing function <lambda>92[1]..<lambda>91[1]..80[1]..iced#nrepl#reconnect[8]..iced#nrepl#connect[24]..69:
line    6:
connection failed: connection refused

```

**Comment by liquidz** [comment]
Currently IcedReconnect will use same port number for reconnecting.
So IcedReconnect cannot use for instant connection for now.

**Comment by tami5** [comment]
Oh I see, so disconnecting and then InstantConnect should work? Let me try

**Comment by tami5** [comment]
Yep works well. Thanks

**Comment by liquidz** [comment]
Closed by #299
